Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2CameraSensorHook.cpp

```diff
@@ -49,14 +33,21 @@ namespace ROS2::SDFormat
                                                          const sdf::Sensor& sdfSensor) -> SensorImporterHook::ConvertSensorOutcome
         {
             auto* cameraSensor = sdfSensor.CameraSensor();
+            if (!cameraSensor)
+            {
+                return AZ::Failure(AZStd::string("Failed to read parsed SDFormat data of %s camera sensor", sdfSensor.Name().c_str()));
+            }
 
             CameraSensorConfiguration cameraConfiguration;
             cameraConfiguration.m_depthCamera = cameraSensor->HasDepthCamera();
             cameraConfiguration.m_colorCamera = (sdfSensor.Type() != sdf::SensorType::DEPTH_CAMERA) ? true : false;
             cameraConfiguration.m_width = cameraSensor->ImageWidth();
             cameraConfiguration.m_height = cameraSensor->ImageHeight();
-            cameraConfiguration.m_verticalFieldOfViewDeg =
-                cameraSensor->HorizontalFov().Degree() * (cameraConfiguration.m_height / cameraConfiguration.m_width);
+            if (cameraConfiguration.m_width != 0)
+            {
+                cameraConfiguration.m_verticalFieldOfViewDeg =
```

Comment:
What should the value of `m_verticalFieldOfViewDeg` be if the `cameraConfiguration.m_width` is zero?

Reply:
Mathematically vertical FoV for image width equal to zero would be infinity. In the practical case camera with zero width (or height) makes no sense. The parameters (including FoV) are checked [later in the pipeline](https://github.com/o3de/o3de-extras/blob/223269b7f1245ea26a87f2efa3f3ac337ea0035d/Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp#L36-L49) - this test is added to avoid software crashes while importing robots with improper parameters.